src/autometrics/decorator.py

```diff
@@ -130,8 +144,14 @@ async def async_wrapper(*args: P.args, **kwds: P.kwargs) -> T:
         async_wrapper.__doc__ = append_docs_to_docstring(func, func_name, module_name)
         return async_wrapper
 
+    def pick_decorator(func: Callable) -> Callable:
```

Comment:
ahhhh nice fix

Reply:
i added another test to catch this issue in 2051c519fcc6040b04962620707fc4c442d2aab2